test/common/router/config_impl_test.cc

```diff
@@ -4137,6 +4137,26 @@ TEST_F(RoutePropertyTest, TestRouteCorsLegacyConfig) {
   EXPECT_EQ(cors_policy->allowCredentials(), true);
 }
 
+TEST_F(RoutePropertyTest, TestBadCorsConfig) {
+  const std::string yaml = R"EOF(
+virtual_hosts:
+- name: default
+  domains:
+  - "*"
+  routes:
+  - match:
+      prefix: "/api"
+    route:
+      cluster: ats
+      cors:
+        enabled: 0
+        allow_credentials: true
+)EOF";
+
+  EXPECT_THROW(TestConfigImpl(parseRouteConfigurationFromV2Yaml(yaml), factory_context_, true),
```

Comment:
Can you potentially make this an EXPECT_THROW_WITH_MESSAGE to make this a slightly better test?